flink-table-store-core/src/main/java/org/apache/flink/table/store/file/KeyValueSerializer.java

```diff
@@ -45,6 +49,13 @@ public class KeyValueSerializer extends ObjectSerializer<KeyValue> {
     private final OffsetRowData reusedValue;
     private final KeyValue reusedKv;
 
+    private TableSchema tableSchema;
+
+    public KeyValueSerializer(RowType keyType, RowType valueType, TableSchema tableSchema) {
```

Comment:
I think we can introduce a special deserializer for `ColumnarRowIterator`.
After getting `VectorizedColumnBatch` from `ColumnarRowData`. We can get `ColumnVector[]`. Then, we can do our projection to produce key and value for `KeyValue`.